Index: includes/menu.inc
===================================================================
RCS file: /cvs/drupal/drupal/includes/menu.inc,v
retrieving revision 1.145
diff -u -r1.145 menu.inc
--- includes/menu.inc	2 Jan 2007 05:05:38 -0000	1.145
+++ includes/menu.inc	6 Jan 2007 17:57:59 -0000
@@ -1129,6 +1129,9 @@
 
       // If the administrator has changed the item, reflect the change.
       if ($item->type & MENU_MODIFIED_BY_ADMIN) {
+        if (module_exists('menu_per_role') && !is_null($access = menu_per_role_access($item->mid))) {
+          $_menu['items'][$item->mid]['access'] = isset($_menu['items'][$item->mid]['access']) ? $_menu['items'][$item->mid]['access'] && $access : $access;
+        }
         $_menu['items'][$item->mid]['title'] = $item->title;
         $_menu['items'][$item->mid]['description'] = $item->description;
         $_menu['items'][$item->mid]['pid'] = $item->pid;
